certbot-nginx/certbot_nginx/parser.py

```diff
@@ -125,8 +148,20 @@ def get_vhosts(self):
                                         path)
                 vhosts.append(vhost)
 
+        self._update_vhosts_addrs_ssl(vhosts)
+
         return vhosts
 
+    def _update_vhosts_addrs_ssl(self, vhosts):
+        """Update a list of raw parsed vhosts to include global address sslishness
+        """
+        addr_to_ssl = self._build_addr_to_ssl()
+        for vhost in vhosts:
+            for addr in vhost.addrs:
+                addr.ssl = addr_to_ssl[addr.normalized_tuple()]
+                if addr.ssl:
+                    vhost.ssl = True
```

Comment:
It seems to me that this implies that if I have one vhost/server block X listening on addresses A and B, and another server block Y listening on addresses B and C, and the first server block includes a `Listen 443` directive, then we will have `Y.ssl = True`, and we'll reason about it as though the C socket is also SSLish.  Am I reading that correctly, and is that that the correct behaviour?